drop-transfer/src/check.rs

```diff
@@ -7,51 +7,91 @@ use tokio_util::sync::CancellationToken;
 use crate::{protocol, service::State, tasks::AliveGuard, IncomingTransfer, Transfer};
 
 pub(crate) fn spawn(
+    refresh_trigger: tokio::sync::watch::Receiver<()>,
     state: Arc<State>,
     xfer: Arc<IncomingTransfer>,
     logger: Logger,
     guard: AliveGuard,
     stop: CancellationToken,
-) -> tokio::task::JoinHandle<()> {
+) {
     let id = xfer.id();
-    let job = run(state, xfer, logger.clone());
 
     tokio::spawn(async move {
         let _guard = guard;
+        let mut refresh_trigger = refresh_trigger;
```

Comment:
```suggestion
```
and change the `spawn()` function parameter to be mutable `mut refresh_trigger: tokio::sync::watch::Receiver<()>,`

Reply:
that's nice, I was actually writing `a: mut T` and it wasn't compiling initially :D looks like the `mut` needs to appear on the left side